packages/amplify-go-function-runtime-provider/resources/localinvoke/main.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/rpc"
+	"os"
+	"time"
+
+	"github.com/aws/aws-lambda-go/lambda/messages"
+	lc "github.com/aws/aws-lambda-go/lambdacontext"
+)
+
+// Code is based on https://github.com/yogeshlonkar/aws-lambda-go-test
+
+const rpcFunctionName = "Function.Invoke"
+const connectionDelay = time.Duration(100) * time.Millisecond
+const maxConnectionRetries = 20
+
+// This is the expected type from the CLI
+type Envelope struct {
+	TimeoutMilliseconds int
+	Port                int
+	Payload             string
+}
+
+type Result struct {
+	Response string
+	Error    string
+}
+
+type LambdaInput struct {
+	Timeout       time.Duration
+	Port          int
+	Payload       interface{}
+	ClientContext *lc.ClientContext
+	Deadline      *messages.InvokeRequest_Timestamp
+}
+
+func invokeLambda(input LambdaInput) ([]byte, error) {
+	// 1. Wait until Lambda's port can be connected, retry opening RPC channel
+	connectionRetries := 1
+	connected := false
+
+	var connection *rpc.Client
+	var err error
+
+	for {
+		connection, err = rpc.Dial("tcp", fmt.Sprintf(":%d", input.Port))
+
+		if connection != nil {
+			connected = true
+			break
+		}
+
+		connectionRetries++
+		time.Sleep(connectionDelay)
+
+		if connectionRetries > maxConnectionRetries {
```

Comment:
nit can move this check above the sleep to avoid an unnecessary sleep at the end